src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -637,7 +637,8 @@ internal static void GetBytes(in decimal d, Span<byte> buffer)
 
         internal static decimal ToDecimal(ReadOnlySpan<byte> span)
         {
-            Debug.Assert(span.Length >= 16);
+            span = span.Slice(0, 16);
+
             int lo = BinaryPrimitives.ReadInt32LittleEndian(span);
             int mid = BinaryPrimitives.ReadInt32LittleEndian(span.Slice(4));
             int hi = BinaryPrimitives.ReadInt32LittleEndian(span.Slice(8));
```

Comment:
should we just `int flags = BinaryPrimitives.ReadInt32LittleEndian(span.Slice(12));` put first?